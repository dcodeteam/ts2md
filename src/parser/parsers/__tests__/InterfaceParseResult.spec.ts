import { parseFixtures } from "../../../__tests__/parseFixtures";

parseFixtures("interface");
