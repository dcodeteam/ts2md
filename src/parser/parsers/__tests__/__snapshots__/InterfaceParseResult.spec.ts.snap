// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface export-default-interface.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": true,
      "documentation": "",
      "documentationTags": Array [],
      "exported": true,
      "extendedInterfaces": Array [],
      "id": "ExportDefaultInterface",
      "methods": Array [],
      "properties": Array [],
    },
  ],
}
`;

exports[`interface export-interface.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": true,
      "extendedInterfaces": Array [],
      "id": "ExportInterface",
      "methods": Array [],
      "properties": Array [],
    },
  ],
}
`;

exports[`interface interface-constructor.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [
        ConstructorParseResult {
          "accessibility": "public",
          "documentation": "",
          "documentationTags": Array [],
          "id": "constructor",
          "parameters": Array [
            ParameterParseResult {
              "documentation": "",
              "documentationTags": Array [],
              "id": "a",
              "type": "number",
            },
            ParameterParseResult {
              "documentation": "",
              "documentationTags": Array [],
              "id": "b",
              "type": "string",
            },
          ],
        },
      ],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": false,
      "extendedInterfaces": Array [],
      "id": "InterfaceConstructor",
      "methods": Array [],
      "properties": Array [],
    },
  ],
}
`;

exports[`interface interface-extends.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": false,
      "extendedInterfaces": Array [],
      "id": "Foo",
      "methods": Array [],
      "properties": Array [],
    },
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": false,
      "extendedInterfaces": Array [],
      "id": "Bar",
      "methods": Array [],
      "properties": Array [],
    },
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": false,
      "extendedInterfaces": Array [
        "Foo",
        "Bar",
      ],
      "id": "Baz",
      "methods": Array [],
      "properties": Array [],
    },
  ],
}
`;

exports[`interface interface-methods.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": true,
      "extendedInterfaces": Array [],
      "id": "InterfaceMethods",
      "methods": Array [
        MethodParseResult {
          "accessibility": "public",
          "documentation": "",
          "documentationTags": Array [],
          "id": "a",
          "parameters": Array [],
          "returnType": "void",
        },
        MethodParseResult {
          "accessibility": "public",
          "documentation": "",
          "documentationTags": Array [],
          "id": "b",
          "parameters": Array [
            ParameterParseResult {
              "documentation": "",
              "documentationTags": Array [],
              "id": "a",
              "type": "string",
            },
          ],
          "returnType": "string",
        },
        MethodParseResult {
          "accessibility": "public",
          "documentation": "",
          "documentationTags": Array [],
          "id": "c",
          "parameters": Array [
            ParameterParseResult {
              "documentation": "",
              "documentationTags": Array [],
              "id": "a",
              "type": "boolean",
            },
            ParameterParseResult {
              "documentation": "",
              "documentationTags": Array [],
              "id": "b",
              "type": "number",
            },
          ],
          "returnType": "string",
        },
      ],
      "properties": Array [],
    },
  ],
}
`;

exports[`interface interface-properties.ts 1`] = `
Object {
  "nodes": Array [
    InterfaceParseResult {
      "constructors": Array [],
      "defaultExported": false,
      "documentation": "",
      "documentationTags": Array [],
      "exported": true,
      "extendedInterfaces": Array [],
      "id": "InterfaceProperties",
      "methods": Array [],
      "properties": Array [
        PropertyParseResult {
          "documentation": "",
          "documentationTags": Array [],
          "id": "a",
          "type": "string",
        },
        PropertyParseResult {
          "documentation": "",
          "documentationTags": Array [],
          "id": "b",
          "type": "string | number",
        },
        PropertyParseResult {
          "documentation": "",
          "documentationTags": Array [],
          "id": "c",
          "type": "(a: string, b: { c: number; }) => boolean",
        },
      ],
    },
  ],
}
`;
